app/src/main/java/ch/epfl/cs311/wanderwave/viewmodel/ProfileViewModel.kt

```diff
@@ -115,8 +114,8 @@ constructor(
    * @since 2.0
    * @last update 2.0
    */
-  fun retrieveTracks() {
-    CoroutineScope(Dispatchers.IO).launch {
+  fun retrieveTracks(scope: CoroutineScope) {
+    scope.launch {
```

Comment:
Is it important that you send the scope ? Seems a bit weird that you control this from the other screen. This is just a question, there probably is a reason, but just to know